website/content/docs/rfcs/namespace-sealing.mdx

```diff
@@ -0,0 +1,421 @@
+  http://127.0.0.1:8200/v1/sys/namespaces/sealednamespace1/unseal
+{ shamir tokens … }
+```
+
+</details>
+
+- Creating a sealed namespace requires a root key. After creation, (un)sealing of the namespace requires the namespace key. A root key can not (un)seal.
+
+#### Retrieve Sealing status of a Namespace
+
+Note: contrary to existing /sys/seal-status, the namespace specific endpoint is authenticated.
+
+| **Method** | **Path**           |
+| ---------- | ------------------ |
+| `GET`      | `/sys/namespaces/:path/seal-status` |
+
+<details>
+
+<summary>Collapse Example</summary>
+
+##### Example
+
+```bash
+$ curl \
+  --header "X-Vault-Token: ..." \
+  --request GET \
+  http://127.0.0.1:8200/v1/sys/namespaces/sealednamespace1/seal-status
+```
+
+</details>
+
+##### Additional Endpoints
+
+Furthermore, the following endpoints will need to be adjusted, so that they are (optionally) namespace aware, by providing X-Vault-Namespace and X-Vault-Token headers.
+
+- `/sys/generate-recovery-token`
+- `/sys/generate-root`
+- `/sys/key-status`
+- `/sys/rekey`
+- `/sys/rekey-recovery-key`
+- `/sys/rotate`
+- `/sys/rotate/config`
+- `/sys/seal`
+- `/sys/seal-status`
+- `/sys/unseal`
+
+### Policies
+
+We probably need to extend Policies, so that we can configure various permissions to for namespace specific actions like sealing, rekeying, key rotation of a namespace.
+
+
+## Rationale and alternatives
+
+Per-namespace encryption moves parts of the responsibility away from the OpenBao operator and closer to the tenant, giving more granular control over its data.
+
+There are no direct alternatives to this feature proposal. It must be highlighted that a feature of (un)locking of a namespace alone is not an alternative to this proposal, as data sovereignty can’t be achieved. However, it could be a first step towards this direction and could be considered a preparation.
+
+### Parallel Unseal
+
+[RFC #1021](https://github.com/openbao/openbao/issues/1021) discussed the a Parallel Unsealing mechanisms to support multiple unseal factors such as Shamirs and KMS. As the RFC only addresses issues of global unsealing, it could be either extended or a new RFC can be written to account for parallel unsealing of Namespaces.
+
+### Lazy Loading of Mounts, Namespaces
+
+Coupled with per-namespace seal mechanisms, one issue with this is that it requires all devices to be online at the time of cluster restart. Instead, one could imagine swapping loading of mounts to be lazily done: this might allow the cluster to start if certain tenants do not have online KMS or HSMs at the time of cluster start, but puts greater availability requirements on these systems in general.
+
+However, this work is strictly disjoint from implementing per-namespaces sealing as a feature and so will be done later.
+
+
+### Downsides
+
+This will add another complex feature to OpenBao; however, this substantially increases competitiveness in the broader Vault Enterprise ecosystem.
```

Comment:
```suggestion
This will add another complex feature to OpenBao; however, this substantially increases competitiveness in the broader Vault Enterprise ecosystem and increase the security posture for tenant separation.
```